src/Silicon.cpp

```diff
@@ -959,39 +1322,210 @@ namespace galsim {
                 dbg<<"ix,iy = "<<ix<<','<<iy<<"  x,y = "<<x<<','<<y<<std::endl;
                 set_verbose(2);
                 bool off_edge;
-                insidePixel(ix, iy, x, y, zconv, target, &off_edge);
-                searchNeighbors(*this, ix, iy, x, y, zconv, target, step);
+                insidePixel(ix, iy, x, y, zconv, target, &off_edge,
+                            emptypolySize, pixelInnerBoundsData,
+                            pixelOuterBoundsData,
+                            horizontalBoundaryPointsData,
+                            verticalBoundaryPointsData,
+                            emptypolyData);
+                searchNeighbors(*this, ix, iy, x, y, zconv, b, step, emptyPolySize,
+                                pixelInnerBoundsData, pixelOuterBoundsData,
+                                horizontalBoundaryPointsData,
+                                verticalBoundaryPointsData, emptyPolyData);
                 set_verbose(1);
 #endif
-                int n = (pixelNotFoundRandom[i-i1] > 0.5) ? 0 : step;
+                const int xoff[9] = {0,1,1,0,-1,-1,-1,0,1}; // Displacements to neighboring pixels
+                const int yoff[9] = {0,0,1,1,1,0,-1,-1,-1}; // Displacements to neighboring pixels
+                int n = (randomArray[(i-i1) * 4 + 2] > 0.5) ? 0 : step;
                 ix = ix + xoff[n];
                 iy = iy + yoff[n];
             }
 
-            if (b.includes(ix,iy)) {
+            if ((ix >= b.getXMin()) && (ix <= b.getXMax()) &&
+                (iy >= b.getYMin()) && (iy <= b.getYMax())) {
```

Comment:
Another place where you replaced includes.  Why can't you just use includes here?